packages/replicache/src/replicache-impl.ts

```diff
@@ -1470,8 +1476,17 @@ export class ReplicacheImpl<MD extends MutatorDefs = {}> {
       );
 
       if (trackingData) {
-        (result as unknown as Record<string, unknown>).server =
-          trackingData.serverPromise;
+        return {
+          client: result,
+          server: trackingData.serverPromise,
+          then: (onFulfilled, onRejected) => {
```

Comment:
catch and finally intentionally left off. The user will get type errors if they try to call those directly. They'll get type errors trying to use `then` too but not with `await` (which calls then).